src/main/java/com/feedzai/commons/sql/abstraction/batch/AbstractPdbBatch.java

```diff
@@ -72,4 +72,29 @@ protected void processBatch(final DatabaseEngine de, final List<BatchEntry> batc
         de.flush();
         de.commit();
     }
+
+    /**
+     * Processes all batch entries ignoring duplicate entries.
+     *
+     * @implSpec Same as {@link #processBatch(DatabaseEngine, List)}}.
+     *
+     * @param de                        The {@link DatabaseEngine} on which to perform the flush.
+     * @param batchEntries              The list of batch entries to be flushed.
+     * @throws DatabaseEngineException  If the operation failed.
+     */
+    protected void processBatchUpsert(final DatabaseEngine de, final List<BatchEntry> batchEntries) throws DatabaseEngineException {
+        /*
+         Begin transaction before the addBatch calls, in order to force the retry of the connection if it was lost during
+         or since the last batch. Otherwise, the addBatch call that uses a prepared statement will fail.
+         */
+        de.beginTransaction();
+
+        for (final BatchEntry entry : batchEntries) {
+            de.addBatchUpsert(entry.getTableName(), entry.getEntityEntry());
+        }
+
+        de.flushUpsert();
+        de.commit();
+    }
+
```

Comment:
DRY - this is duplicated code and should be improved.

(if there is time before the release, I would not make it a priority, just letting a note)

Reply:
This requires some further analysis, right off the bat I can't change it as from what I saw it could take some refactoring.

Reply:
ack! - we can address this later.